src/main/java/tr/edu/ogu/ceng/advice/ImsAOPInternship.java

```diff
@@ -0,0 +1,56 @@
+package tr.edu.ogu.ceng.advice;
+
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import lombok.extern.slf4j.Slf4j;
+import tr.edu.ogu.ceng.dao.CompanyRepository;
+import tr.edu.ogu.ceng.dao.FacultySupervisorRepository;
+import tr.edu.ogu.ceng.dao.InternshipRepository;
+import tr.edu.ogu.ceng.dao.StudentRepository;
+import tr.edu.ogu.ceng.dto.InternshipDto;
+import tr.edu.ogu.ceng.service.Exception.EntityNotFoundException;
+import tr.edu.ogu.ceng.service.Exception.ServiceException;
+
+@Slf4j
+@Aspect
+@Component
+public class ImsAOPInternship {
+
+	@Autowired
+	private InternshipRepository internshipRepository;
+	@Autowired
+	private StudentRepository studentRepository;
+	@Autowired
+	private FacultySupervisorRepository facultySupervisorRepository;
+	@Autowired
+	private CompanyRepository companyRepository;
+
+	@Before("execution(* tr.edu.ogu.ceng.service.InternshipService.updateInternship(..)) && args(internshipDto)")
+	public void beforeUpdateInternship(InternshipDto internshipDto) {
+		boolean existingInternshipId = internshipRepository.existsById(internshipDto.getId());
+		boolean existingFacultySupervisorId = facultySupervisorRepository
+				.existsById(internshipDto.getFacultySupervisorId());
+		boolean existingStudentId = studentRepository.existsById(internshipDto.getStudentId());
+		boolean existingCompanyId = companyRepository.existsById(internshipDto.getCompanyId());
+
+		if (!existingInternshipId) {
+			log.warn("Internship not found with id = {}!", internshipDto.getId());
+			throw new EntityNotFoundException();
+		} else if (!existingFacultySupervisorId) {
+			throw new ServiceException("Fakülte sorumlusu bulunamadı!"); // TODO - EntityNotFoundException with message.
+		} else if (!existingStudentId) {
+			throw new ServiceException("Öğrenci bulunamadı!"); // TODO - EntityNotFoundException with message.
```

Comment:
Bu kısımda fakulte sorumlusu, öğrenci ve şirket için güncellenme aşamasında veri tabanında bulunup bulunmadıklarını kontrol ediyorum. Bu kontrolü yapmazsam veri tabanından otomatik exception atıyor. Sizce bu kontrolleri yapmalı mıyım? Yoksa veri tabanına mı bırakmalıyım? Yorumunuzu bekliyorum.

Reply:
Ben de exceptionlar ile burada ilgilenmenin daha doğru bir yaklaşım olduğunu düşünüyorum.

Reply:
Bence de veri tabanına bırakılmamalı, kontrolleri yapmak kullanıcının hatayı yönetmesi açısından daha verimli olur.